### PR TITLE
chore(toh-5): improve e2e test coverage and readability (use async/await)

### PR DESCRIPTION
**Pull request by chalin**
Improved test coverage. Also updated tests to make use of async/await (so tests are more readable).

This is part of work to get a full suite of e2e tutorial tests to run for Dart (#1619).

### CODE REVIEW

**Comment by chalin** [comment]
@wardbell @Foxandxss @filipesilva : ready for review.

cc @kwalrath

File: public/docs/_examples/toh-5/e2e-spec.ts [hunk 1]
```diff
@@ -1,113 +1,190 @@
...
+      element(by.buttonText('View Details')).click();
+
+      let page = getPageElts();
+      expect(page.heroDetail.isPresent()).toBeTruthy('shows hero detail');
+      let hero = await Hero.fromDetail(page.heroDetail);
+      expect(hero.id).toEqual(targetHero.id);
+      expect(hero.name).toEqual(targetHero.name);
+    });
+
+    it(`updates hero name (${newHeroName}) in details view`, updateHeroNameInDetailView);
```
Comment (Foxandxss) [inline public/docs/_examples/toh-5/e2e-spec.ts]: Isn't this the same test as line 98?

**Comment by Foxandxss** [comment]
Leaving the comment aside, I don't dislike it, the new Hero class makes it read better.

File: public/docs/_examples/toh-5/e2e-spec.ts (hunk 1 above)
Comment (chalin) [inline public/docs/_examples/toh-5/e2e-spec.ts]: Yes, we test the updating of the hero name twice: once when accessed via the Dashboard, and once when accessed via the Heroes list. This is how it was in the previous e2e tests.

Comment (Foxandxss) [inline public/docs/_examples/toh-5/e2e-spec.ts]: But I can't see how they can do anything different. Tests runs from a clean state and they seem to be run from the exact same page.

Shouldn't it be dropped? Or there is something I cannot see?

Comment (chalin) [inline public/docs/_examples/toh-5/e2e-spec.ts]: The test groups for the Dashboard and the Heroes list are **independent**. I think that it is reasonable to include this single test (editing the hero name), in both test groups.

**Pull request merged**